tf_agents/cogment_verse_tf_agents/reinforce/reinforce.py

```diff
@@ -93,31 +95,29 @@ def __loss(prob, actions, Q):
         loss = -log_prob * Q
         return tf.reduce_mean(loss)
 
-    def learn(self, batch, update_schedule=True):
+    def learn(self):
 
+        batch = self.sample_training_batch()
         Q = self.get_discounted_rewards(batch["rewards"])
         with tf.GradientTape() as tape:
             prob = self._model.model(batch["observations"], training=True)
-            prob = self.regularize_dist(prob)
             loss = self.__loss(prob, batch["actions"], Q)
         gradients = tape.gradient(loss, self._model.trainable_variables)
 
         self._optimizer.apply_gradients(zip(gradients, self._model.trainable_variables))
-
-        return {"loss": loss}
+        self.reset_replay_buffer()
+        return {"loss": loss, "num_samples_seen": batch["observations"].shape[0],
+                "rewards_mean": batch["rewards"].mean(), "done_mean": batch["done"].mean()}
 
     def consume_training_sample(self, sample):
         """
         Consume a training sample, e.g. store in an internal replay buffer
         """
         self._replay_buffer.add(sample)
 
-    def sample_training_batch(self, batch_size=32):
+    def sample_training_batch(self, batch_size=64):
```

Comment:
It looks like it's only used by this class, make it a "private" method to make sure it is clear
```suggestion
    def _sample_training_batch(self, batch_size=64):
```